httpc.go

```diff
@@ -303,60 +303,68 @@ var (
 	errorEncodedRawBodyCollision = errors.New("cannot use both body encoding and raw body content")
 )
 
-func (r *Request) prepBody() (body io.ReadCloser, err error) {
+func (r *Request) setBody(req *http.Request) (err error) {
 
 	// If requested, parse the requst body using the specified encoder
+	bodyBytes := r.body
 	if r.bodyEncoder != nil {
 		if len(r.body) > 0 {
-			return body, errorEncodedRawBodyCollision
+			return errorEncodedRawBodyCollision
 		}
 
-		r.body, err = r.bodyEncoder.Encode()
+		bodyBytes, err = r.bodyEncoder.Encode()
 		if err != nil {
-			return body, fmt.Errorf("error encoding body: %w", err)
+			return fmt.Errorf("error encoding body: %w", err)
 		}
+		req.Header.Set("Content-Type", r.bodyEncoder.ContentType())
```

Comment:
Could this conflict with someone manually setting a `Content-Type` header (not sure about the order off the top of my head)?